File: 680-valid-palindrome-ii/680-valid-palindrome-ii.java
class Solution {
    public boolean validPalindrome(String s) {
        int left = 0;
        int right = s.length() - 1;
        while(left<right){
            if (s.charAt(left) != s.charAt(right)){
                String lside  = s.substring(left,right);
                String rside = s.substring(left+1,right+1);
                return (isPalindrome(lside) || isPalindrome(rside));}
            left += 1;
            right -= 1;   
        }
        return true;
    }
    
    public static boolean isPalindrome(String str){
        int left = 0;
        int right = str.length() - 1;
        while(left<right){
            if (str.charAt(left) != str.charAt(right)){
                return false;}
            left += 1;
            right -= 1;   
        }
        return true;
    }
 }

File: 680-valid-palindrome-ii/680-valid-palindrome-ii.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        left = 0
        right = len(s) -1
        while left < right:
            if s[left] != s[right]:
                lside = s[left:right] 
                rside = s[left + 1:right + 1]
                return lside == lside[::-1] or rside == rside[::-1]
            left += 1
            right -= 1
        
        return True
                
            